src/ui/pages/SetPasscode/SetPasscode.tsx

```diff
@@ -42,11 +40,16 @@ const SetPasscode = () => {
     updateReduxState(nextPath.pathname, data, dispatch, updateRedux);
     ionRouter.push(nextPath.pathname, "forward", "push");
 
-    PreferencesStorage.set(PreferencesKeys.APP_ALREADY_INIT, {
-      initialized: true,
-    }).catch((e) => {
-      showError("Unable to save app init state", e, dispatch);
-    });
+    Agent.agent.basicStorage
+      .save({
+        id: MiscRecordId.APP_ALREADY_INIT,
+        content: {
+          initialized: true,
+        },
+      })
+      .catch((e) => {
+        showError("Unable to save app init state", e, dispatch);
+      });
```

Comment:
@Sotatek-DukeVu @sdisalvo-crd This is a perfect example of what I have been saying about catching errors. Here, we have continued to the new path after setting the passcode, and when we fail to store `APP_ALREADY_INIT` we just show a toast.

Now when the user reloads the app, their secure storage will be completely wiped because `APP_ALREADY_INIT` was never set. We NEED to consider these scenarios going forward and be careful with how we catch errors.

@Sotatek-TungNguyen2a I'd suggest you move this call to `basicStorage` to before the router.push call add an `await`. We should not call `ionRouter.push` if an error is thrown

Reply:
So ideally, show the toast (so the user has feedback) and then re-throw the error